pkg/operator/controller/vulnerabilityreport.go

```diff
@@ -108,6 +110,27 @@ func (r *VulnerabilityReportReconciler) reconcileWorkload(workloadKind kube.Kind
 			}
 		}
 
+		if r.Config.VulnerabilityScannerLatest && workloadKind == kube.KindReplicaSet {
+			controller := metav1.GetControllerOf(workloadObj)
+			if controller != nil && controller.Kind == string(kube.KindDeployment) {
+				deploymentPartialObject := kube.Object{
+					Kind:      kube.KindDeployment,
+					Name:      controller.Name,
+					Namespace: req.Namespace,
+				}
+				deploymentObject, err := r.GetObjectFromPartialObject(ctx, deploymentPartialObject)
+				if err != nil {
+					return ctrl.Result{}, err
+				}
+				deploymentRevisionAnnotation := deploymentObject.GetAnnotations()
+				replicasetRevisionAnnotation := workloadObj.GetAnnotations()
+				if replicasetRevisionAnnotation[deploymentAnnotation] != deploymentRevisionAnnotation[deploymentAnnotation] {
```

Comment:
I think we could extract this logic to a new method defined on the ObjectResolver struct, i.e. `ObjectResolver.IsActiveReplicaSet(workloadObj)` or `ObjectResolver.IsCurrentRevision(workloadObj)`. Then you could write simple unit tests for this method.

NB The `kube.ObjectResolver` instance is already injected to the controller and has defined similar methods to look up Deployment by ReplicaSet or ReplicaSet by Pods.